File: 313B.java
import java.util.Scanner;
public class IliyaQuery
{
public static void main(String ar[]){
Scanner sb=new Scanner(System.in);
String s=sb.nextLine();
String s1=sb.nextLine();
int m=Integer.parseInt(s1);
int a[][]=new int[m][2];
int c=0;
for(int i=0;i<m;i++) {
   String p=sb.nextLine();
   String p1[]=p.split(" ");
   a[m][0]=Integer.parseInt(p1[0]);
   a[m][1]=Integer.parseInt(p1[1]);
    }
	/*for(int h=0;h<m;h++) {
   for(int j=a[h][0]-1;j<a[h][1]-1;j++) {
           if(s.charAt(j)==s.charAt(j+1)) {c++;}
		   }
		   System.out.println(c);
		   c=0;
				}*/
				}
				}
				

File: 313B.py
s=raw_input("")
k=int(raw_input(""))
l=[0]
for i in range(1,len(s)):
	if s[i]==s[i-1]:
		l.append((l[i-1]+1))
	else:
		l.append(l[i-1])
r=[]
for i in range(k):
	q=map(int,raw_input("").split(" "))
	r.append((l[q[1]-1]-l[q[0]-1]))
for i in r:
	print i
